test/hotspot/jtreg/runtime/exceptionMsgs/NoClassDefFoundError/libNoClassDefFoundErrorTest.c

```diff
@@ -42,3 +46,34 @@ Java_NoClassDefFoundErrorTest_callFindClass(JNIEnv *env, jclass klass, jstring c
 }
 
 
+static char* giant_string() {
+    size_t len = ((size_t)INT_MAX) + 3;
+    char* c_name = malloc(len * sizeof(char));
+    if (c_name != NULL) {
+        memset(c_name, 0x59595959, len - 1); // YYYY...
+        c_name[len - 1] = '\0';
+    }
+    return c_name;
+}
+
+JNIEXPORT jboolean JNICALL
+Java_NoClassDefFoundErrorTest_tryCallDefineClass(JNIEnv *env, jclass klass) {
+    char* c_name = giant_string();
+    if (c_name != NULL) {
+        (*env)->DefineClass(env, c_name, NULL, NULL, 0);
+        free(c_name);
+        return JNI_TRUE;
+    }
+    return JNI_FALSE;
```

Comment:
You don't free the class name so that the memory is available to the error message?  Maybe a comment would be useful here and below.

Reply:
Since the test is run with othervm, the memory should be reclaimed with the process exit.  Probably no comment needed since I think we can't run native code without othervm.

Reply:
We can run native code without othervm but I switched to othervm both for the logging and to make it more likely we would not be short on memory.

The malloc'd name is used to construct the error message but not used as part of the error message so can be freed once the JNI call is complete.

Reply:
othervm seems like a good choice for this but now when you return JNI_FALSE, where do you free the string?

Reply:
If we return false then the malloc failed.